content/news/2020-06-19-gsoc-update2-2020.md

```diff
@@ -4,12 +4,14 @@ date = 2020-06-19
 extra.author = "Devin Ragotzy"
 +++
 
-This week in ruma-events' Google Summer of Code project, after trying out the new API in [`matrix-rust-sdk`](https://github.com/DevinR528/matrix-rust-sdk/pull/2), we found a few fatal flaws. There was no easy way to go from `StateEvent<AnyStateEventContent>` to `StateEvent<SpecificEventContent>`, the other issue was one could create a `StateEvent` with differing `content` and `prev_content` fields using the `AnyStateEventContent` enum. The [new API](https://github.com/ruma/ruma/pull/68) will be similar to the old one, each event kind (state, message, ephemeral, etc.) is represented by an enum.
+This week in ruma-events' Google Summer of Code project, after trying out the new `Any*Event` enums [`matrix-rust-sdk`](https://github.com/DevinR528/matrix-rust-sdk/pull/2), we found a few big flaws. There was no easy way to go from `StateEvent<AnyStateEventContent>` to `StateEvent<SpecificEventContent>`, the other issue was one could create a `StateEvent` with differing `content` and `prev_content` fields using the `AnyStateEventContent` enum. The [0.22 ruma-events](https://github.com/ruma/ruma/pull/68) will be similar to the existing API, each event kind (state, message, ephemeral, etc.) is still a generic struct but, each event type (think "m.room.aliases") will be represented by an enum instead of the `content` field keeping track of the event type.
```

Comment:
> each event type (think "m.room.aliases") will be represented by an enum

Uhh, what?

Reply:
> The [0.22 ruma-events](https://github.com/ruma/ruma/pull/68) will be similar to the existing API, as each event will be represented by an enum with event type variants.

Reply:
So I'll just replace the whole last sentence with the above.

Reply:
That still doesn't sound right. I think I'm looking for 'each event type has its own event enum *variant*' somewhere in there.